packages/lodestar/test/utils/stub/beaconDb.ts

```diff
@@ -0,0 +1,84 @@
+import {SinonSandbox, SinonStubbedInstance} from "sinon";
+
+import {config as mainnetConfig} from "@chainsafe/lodestar-config/lib/presets/mainnet";
+import {SignedBeaconBlock, BeaconState, BLSPubkey, ValidatorIndex} from "@chainsafe/lodestar-types";
+
+import {BeaconDb, LevelDbController} from "../../../src/db";
+import {
+  AttesterSlashingRepository,
+  AggregateAndProofRepository,
+  AttestationRepository,
+  BadBlockRepository,
+  BlockRepository,
+  BlockArchiveRepository,
+  ChainRepository,
+  DepositDataRepository,
+  DepositDataRootListRepository,
+  ProposerSlashingRepository,
+  StateRepository,
+  VoluntaryExitRepository,
+} from "../../../src/db/api/beacon/repositories";
+
+export class StubbedBeaconDb extends BeaconDb {
+  public db: SinonStubbedInstance<LevelDbController>;
+
+  public chain: SinonStubbedInstance<ChainRepository> & ChainRepository;
+  public state: SinonStubbedInstance<StateRepository> & StateRepository;
+  public badBlock: SinonStubbedInstance<BadBlockRepository> & BadBlockRepository;
+  public block: SinonStubbedInstance<BlockRepository> & BlockRepository;
+  public blockArchive: SinonStubbedInstance<BlockArchiveRepository> & BlockArchiveRepository;
+
+  public attestation: SinonStubbedInstance<AttestationRepository> & AttestationRepository;
+  public aggregateAndProof: SinonStubbedInstance<AggregateAndProofRepository> & AggregateAndProofRepository;
+  public voluntaryExit: SinonStubbedInstance<VoluntaryExitRepository> & VoluntaryExitRepository;
+  public proposerSlashing: SinonStubbedInstance<ProposerSlashingRepository> & ProposerSlashingRepository;
+  public attesterSlashing: SinonStubbedInstance<AttesterSlashingRepository> & AttesterSlashingRepository;
+  public depositData: SinonStubbedInstance<DepositDataRepository> & DepositDataRepository;
+
+  public depositDataRootList: SinonStubbedInstance<DepositDataRootListRepository> & DepositDataRootListRepository;
+
+  public storeChainHead:
+  SinonStubbedInstance<(block: SignedBeaconBlock, state: BeaconState) => Promise<void>>
+  &
+  ((block: SignedBeaconBlock, state: BeaconState) => Promise<void>);
+
+  public updateChainHead:
+  SinonStubbedInstance<(blockRoot: Uint8Array, stateRoot: Uint8Array) => Promise<void>>
+  &
+  ((blockRoot: Uint8Array, stateRoot: Uint8Array) => Promise<void>);
+
+  public getValidatorIndex:
+  SinonStubbedInstance<(publicKey: BLSPubkey) => Promise<ValidatorIndex>>
+  &
+  ((publicKey: BLSPubkey) => Promise<ValidatorIndex>);
+
+  public processBlockOperations:
+  SinonStubbedInstance<(signedBlock: SignedBeaconBlock) => Promise<void>>
+  &
+  ((signedBlock: SignedBeaconBlock) => Promise<void>);
+
+  constructor(sinon: SinonSandbox, config = mainnetConfig) {
```

Comment:
Maybe minimal config by default is better as we use minimal in almost all of our unit tests?

Reply:
will update after fixing conflicts / dependency PR